src/pages/Task.tsx

```diff
@@ -282,7 +282,9 @@ const TaskCore: React.FC = () => {
                     bbData={buildBaronData}
                     error={buildBaronError}
                     taskId={id}
+                    execution={execution}
                     loading={buildBaronLoading}
+                    userModifyPermission={annotation?.userModifyPermission}
                   />
```

Comment:
Thinking about how these props are used allot of these props are just being drilled down from the `Task` page to underlying build baron components and theres allot of build baron specific code in this file. What do you think about making a `BuildBaron` tab component that handles all of this so we can simplify the `Task` page component.  cc. @malikchaya2  @SupaJoon

Reply:
It's all being passed into the <BuildBaron> component which handles everything. As far as passing data through: the annotation is part of the task query, and we need it there because that determines wether we show the tab. Same with the buildBaronData. We need to know if we should show the tab, so we need to do that query here. I don't think we can extract anything....